packages/next/pages/_document.tsx

```diff
@@ -144,24 +144,33 @@ export class Head extends Component<
 
   context!: DocumentComponentContext
 
-  getCssLinks() {
+  getCssLinks(): JSX.Element[] | null {
     const { assetPrefix, files } = this.context._documentProps
     const cssFiles =
       files && files.length ? files.filter(f => /\.css$/.test(f)) : []
+    const cssLinkElements: JSX.Element[] = []
+    cssFiles.forEach(file => {
+      cssLinkElements.push(
```

Comment:
Is this the right place to add the preload?

We'll have links in the order 

`<link rel="preload" href="/_next/static/css/abc.chunk.css" as="style">`
`<link rel="stylesheet" href="/_next/static/css/abc.chunk.css">`

Essentially we don't have it loading any sooner than before? Without the preload it would have encountered the link tag and initiated load anyways.
Let me know if Im missing something

Reply:
Surprisingly this actually has an effect, in Chrome at least. @spanicker tipped me off to it, and I tried it with a local override to confirm. The preload link seems superfluous, but it causes the CSS to go from the bottom of the download list to the top.

Reply:
does the relative ordering of css preload vs. JS preload matter?
Safer to order CSS before JS?

Reply:
I'm guessing order doesn't matter? it sounds like Chrome tries to download CSS before JS, as long as they're the same "type" of download, but that it gives precedence to preloads as a whole class. So if the CSS is also a preload, it'll get prioritized once again. Also, in my testing adding the preload tag in place above the link stylesheet tag had the intended effect

Reply:
Also will this behave the same way in other browsers?